lib/domain.js

```diff
@@ -21,6 +21,10 @@
 
 'use strict';
 
+if (process.isWorkerThread) {
+  throw new Error('domains are not available inside of worker threads');
```

Comment:
That's a pretty big decision to make. Personally I'm +1 on it but it means that any code that currently runs in child processes won't be able to run on workers.

Petka's original idea was to convert `cluster` to use workers.

Reply:
> it means that any code that currently runs in child processes won't be able to run on workers.

You mean “some code”, right? As in, child processes and workers are incompatible? I’m okay with that as well.

> Petka's original idea was to convert `cluster` to use workers.

I would assume by now too much code relies on that not being the case anyway.

Reply:
> You mean “some code”, right? As in, child processes and workers are incompatible? I’m okay with that as well.

That sentence was missing "and uses domains" :)

I think that people currently using child processes to offload work are the most obvious potential consumers of workers and this _might_ make their life harder.

> I would assume by now too much code relies on that not being the case anyway.

I'm not sure why?

Reply:
> > I would assume by now too much code relies on that not being the case anyway.

> I'm not sure why?

For one, the current implementation in this PR doesn’t provide stdio to Workers, but I would be pretty sure a lot of cluster-using code wants that to be there ;)

Reply:
Come to think of it - can't stdio just be done through a MessagePort?

Reply:
@benjamingr Yes … but that could also be done by the user, right? `stdio` in Node/Ayo is complicated, and I’m a bit scared of what happens when multiple handles to the same resource exist (like, some could be set to blocking, data would end up being intertwined, there would be different libuv buffers, and for `stdin` there can’t really be any consistent behaviour anyway.

I’ve made `console.*` work by just using a fully synchronous stream that operates directly on the file descriptors, which might not be super-efficient but I think it’s reasonable here.

Reply:
I think that's reasonable and we can always add it later.